File: backend/app/analytics/schemas/analytics.py
from pydantic import BaseModel


class SystemStats(BaseModel):
    total_users: int
    total_connections: int
    total_jobs: int
    total_posts: int
    total_events: int = 0
    total_clubs: int = 0
    total_likes: int = 0
    total_comments: int = 0
    total_post_views: int = 0
    total_profile_views: int = 0


class PlatformEngagementSummary(BaseModel):
    total_likes: int
    total_comments: int
    total_post_views: int
    total_profile_views: int
    total_engagement_actions: int


class ProfileViewItem(BaseModel):
    date: str
    views: int


class ProfileViewsResponse(BaseModel):
    total_views: int
    history: list[ProfileViewItem]


class PostEngagementItem(BaseModel):
    post_id: int
    content_snippet: str
    created_at: str
    likes: int
    comments: int
    views: int


class PostEngagementResponse(BaseModel):
    total_likes: int
    total_comments: int
    total_views: int
    posts: list[PostEngagementItem]


class TrendingPostResponse(BaseModel):
    post_id: int
    content: str
    created_at: str
    author_name: str
    likes: int
    comments: int
    views: int
    score: int

File: frontend/src/services/analytics.ts
import { apiRequest } from './api'

export interface SystemStats {
  total_users: number
  total_connections: number
  total_jobs: number
  total_posts: number
  total_events?: number
  total_clubs?: number
  total_likes?: number
  total_comments?: number
  total_post_views?: number
  total_profile_views?: number
}

export interface PlatformEngagementSummary {
  total_likes: number
  total_comments: number
  total_post_views: number
  total_profile_views: number
  total_engagement_actions: number
}

export interface ProfileViewItem {
  date: string
  views: number
}

export interface ProfileViewsResponse {
  total_views: number
  history: ProfileViewItem[]
}

export interface PostEngagementItem {
  post_id: number
  content_snippet: string
  created_at: string
  likes: number
  comments: number
  views: number
}

export interface PostEngagementResponse {
  total_likes: number
  total_comments: number
  total_views: number
  posts: PostEngagementItem[]
}

export interface TrendingPost {
  post_id: number
  content: string
  created_at: string
  author_name: string
  likes: number
  comments: number
  views: number
  score: number
}

export const analyticsService = {
  getSystemStats: async (): Promise<SystemStats> => {
    return apiRequest<SystemStats>('/analytics/stats')
  },

  getPlatformEngagementSummary: async (): Promise<PlatformEngagementSummary> => {
    return apiRequest<PlatformEngagementSummary>('/analytics/platform/engagement-summary')
  },

  getProfileViews: async (days: number = 7): Promise<ProfileViewsResponse> => {
    return apiRequest<ProfileViewsResponse>(`/analytics/profile/views?days=${days}`)
  },

  getPostEngagement: async (): Promise<PostEngagementResponse> => {
    return apiRequest<PostEngagementResponse>('/analytics/posts/engagement')
  },

  getTrendingPosts: async (limit: number = 5, days: number = 7): Promise<TrendingPost[]> => {
    return apiRequest<TrendingPost[]>(`/analytics/trending-posts?limit=${limit}&days=${days}`)
  },

  recordPostView: async (postId: number): Promise<void> => {
    return apiRequest<void>(`/analytics/posts/${postId}/view`, {
      method: 'POST',
    })
  },

  recordProfileView: async (profileId: number): Promise<void> => {
    return apiRequest<void>(`/analytics/profile/${profileId}/view`, {
      method: 'POST',
    })
  },
}
